canvas.go

```diff
@@ -249,8 +233,10 @@ c73.985,0,73.985,0,73.985-73.986V72.986C84.979-1,84.979-1,10.994-1z`,
 	for _, relation := range c.relations {
 		relation.definition(canvas)
 	}
+	c.iconsRendered = make(map[string]bool)
+	c.iconIds = make(map[string]string)
```

Comment:
I don't see why these two are added as fields in the Canvas. They don't seem
to be used there. I would suggest either creating them when the canvas
is created (not here) and passing the Canvas into service.definition, or
deleting them from the Canvas struct and keeping things pretty much
as they are. Probably the latter unless there's a compelling case for
the former.